### PR TITLE
Allow setting Guzzle options on a service

### LINKED ISSUE #16: Allow passing Guzzle client options
I want to be able to pass options to the Guzzle client (like HTTP timeout). I'm able to do this on the old SDK by setting up a request filter and doing something like `$request->setConfig('timeout', $timeout)`, but this isn't available anymore.

### PR DESCRIPTION
**Pull request by Sazpaimon**
Should fix #16, at least for everything except for setting options on an individual request

### CODE REVIEW

**Comment by Sazpaimon** [comment]
@nowenL Let me know if you need any changes, I'd like to get this merged ASAP so I can start using it without referencing my fork in my project.

**Comment by Sazpaimon** [comment]
Also, my IDE removes trailing whitespace automatically. I can revert that, or you can view the diff at https://github.com/Azure/azure-storage-php/pull/19/files?w=1 to ignore whitespace from the diff

File: src/Common/Internal/ServiceRestProxy.php [hunk 1]
```diff
@@ -50,78 +50,86 @@ class ServiceRestProxy extends RestProxy
...
      * Initializes new ServiceRestProxy object.
      *
      * @param string      $uri            The storage account uri.
      * @param string      $accountName    The name of the account.
      * @param ISerializer $dataSerializer The data serializer.
+     * @param array       $guzzleOptions  Array of options to pass to the Guzzle client
      */
-    public function __construct($uri, $accountName, $dataSerializer)
-    {           
+    public function __construct($uri, $accountName, $dataSerializer, $guzzleOptions = [])
```
Comment (nowenL) [inline src/Common/Internal/ServiceRestProxy.php]: how about name it 'options' and use 'guzzle' => [] element inside options for guzzle specific settings? Then we can add more general options in the further without adding more parameters.

File: src/Common/Internal/ServiceRestProxy.php [hunk 2]
```diff
@@ -132,58 +140,63 @@ protected function send(
...
-        
+                    ),
+                    'cookies' => true,
+                    'verify' => false,
+                    // For testing with Fiddler
+                    // 'proxy' => "localhost:8888",
+                )
+            )
+        );
+        $client->getConfig();
```
Comment (nowenL) [inline src/Common/Internal/ServiceRestProxy.php]: Why get config here?

File: src/Common/ServicesBuilder.php [hunk 3]
```diff
@@ -183,10 +185,10 @@ public function createQueueService($connectionString)
      * Builds a blob object.
      *
      * @param string $connectionString The configuration connection string.
-     *
+     * @param array  $guzzleOptions    Array of options to pass to Guzzle client
      * @return MicrosoftAzure\Storage\Blob\Internal\IBlob
      */
-    public function createBlobService($connectionString)
+    public function createBlobService($connectionString, $guzzleOptions = [] = [])
```
Comment (nowenL) [inline src/Common/ServicesBuilder.php]: duplicate =[]

**Comment by nowenL** [comment]
Thank you for the contribution! This change looks good and straight forward. I added some comments in the PR. Besides, pls develop based on dev branch instead of master and update changelog.md as well.

**Comment by Sazpaimon** [comment]
Okay, closing this PR in order to make a new one against the dev branch

**Pull request closed without merging**